### PR TITLE
AUT-272: Pass identity claims to SPOT for P2

### PR DESCRIPTION
**Pull request by dbes-gds**
## What?

Pass identity claims to SPOT for P2.

Retrieving the IPV core user identity outputs:

https://github.com/alphagov/digital-identity-architecture/blob/main/rfc/0014-identity-ipv-core-oauth-profile.md

Passing to SPOT:

https://github.com/alphagov/digital-identity-architecture/blob/main/rfc/0013-spot-request-response-structures.md

## Why?

Claims need to be signed by SPOT before being returned to the RP.

### CODE REVIEW

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java [hunk 1]
```diff
@@ -276,4 +294,47 @@ public APIGatewayProxyResponseEvent handleRequest(
...
+        if (ipvCredentials != null && ipvCredentials instanceof ArrayList) {
+            spotClaimsBuilder.withCredentials(
+                    (ArrayList) userIdentityUserInfo.getClaim(IPV_CREDENTIALS.getValue()));
+        }
+
+        var ipvCoreIdentity = userIdentityUserInfo.getClaim(IPV_CORE_IDENTITY.getValue());
+        if (ipvCoreIdentity != null) {
+            spotClaimsBuilder.withIdentityCredentials(
+                    userIdentityUserInfo.getClaim(IPV_CORE_IDENTITY.getValue()));
+        }
```
Comment (JHjava) [inline ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java:327]: This looks good for now. We will need to re-visit this when IPV return all the info as we may not want to send anything to SPOT if credentials are missing from the IPV user-identity response. Maybe add a placeholder jira ticket in slice 2?

**Review by JHjava — COMMENTED** [review]
Looks good 👍  Main query is around managing the hardcoded IdentityClaims. Might need a discussion around that, but this can be iterated on.

File: shared/src/main/java/uk/gov/di/authentication/shared/entity/IdentityClaims.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.di.authentication.shared.entity;
+
+public enum IdentityClaims {
+    VOT("vot"),
+    VTM("vtm"),
+    CREDENTIAL_JWT("https://vocab.account.gov.uk/v1/credentialJWT"),
+    IDENTITY_CREDENTIALS("https://vocab.account.gov.uk/v1/IdentityCredential"),
+    IPV_CREDENTIALS("https://vocab.sign-in.service.gov.uk/v1/credentials"),
+    IPV_CORE_IDENTITY("https://vocab.account.gov.uk/v1/coreIdentity");
```
[the remark this answers is not in the document]
Comment (dbes-gds) [reply shared/src/main/java/uk/gov/di/authentication/shared/entity/IdentityClaims.java:9]: At the moment what's coming back from IPV is not what's in the spec and not necessarily what SPOT is expecting, so have created these to make it work for the time being.  When it stabilises we might be able to remove.  Also if all the claims aren't passed to spot then you need to know which ones to pass, so those will have to be configured somewhere.

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java (hunk 1 above)
Comment (dbes-gds) [reply ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java]: now copying everything across except sub as spot has a separate field for that 'out_sub'.

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java [hunk 3]
```diff
@@ -224,35 +243,34 @@ public APIGatewayProxyResponseEvent handleRequest(
...
-                                                new LogIds(session.getSessionId()));
+
                                 if (configurationService.isSpotEnabled()) {
-                                    sqsClient.send(objectMapper.writeValueAsString(spotRequest));
-                                    LOG.info("SPOT request placed on queue");
+                                    if (LevelOfConfidence.MEDIUM_LEVEL
+                                            .getValue()
+                                            .equals(
+                                                    userIdentityUserInfo.getClaim(
+                                                            VOT.getValue()))) {
```
[the remark this answers is not in the document]
Comment (dbes-gds) [reply ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java]: going to leave as-is as discussed for MVP.

File: shared/src/main/java/uk/gov/di/authentication/shared/entity/IdentityClaims.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbes-gds) [reply shared/src/main/java/uk/gov/di/authentication/shared/entity/IdentityClaims.java]: now just passing on what comes from IPV so have removed constants we don't look at.

**Review by JHjava — APPROVED** [review]

**Pull request merged**